### DISCUSSION TITLE
Use 'd' to toggle debug mode

### DISCUSSION THREAD

**Discussion by adamjstewart** [spack/spack#12206] [Ideas]
Much like you can type <kbd>v</kbd> during `spack install` to see a verbose output of the build, it would be nice to be able to type <kbd>d</kbd> during any Spack command to toggle debug mode. This will help when Spack slows to a halt and you want to make sure it's still doing something.

**Comment by becker33** [comment]
This would require launching a separate process during all Spack invocations just to listen for the `d` key. The implementation on the `v` toggle for verbosity relies on the fact that we already have a logging daemon active during package install processes. I don't know whether this would be worth it given the need for a separate process, what do you think?

**Comment by adamjstewart** [comment]
Probably not, depending on how much resources this process requires...
